From 8f8799cfc64f647d701b41591784d9481762d574 Mon Sep 17 00:00:00 2001
From: Angela Czubak <aczubak@marvell.com>
Date: Mon, 4 Mar 2019 15:06:12 +0100
Subject: [PATCH 0087/1921] octeontx2-af: fix rvu_sso_ggrp_taq_flush

ADD_WORK should refer to slot and not lf.
BAR2 aliasing assumes that we chose which PFVF to refer to and
the unit within it we should work on is a slot.

Change-Id: Iff14a32ecc2fe4946ee457aa3ba6a7bcd707951c
Signed-off-by: Angela Czubak <aczubak@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
index 4054c5261515..10bafdac3fbf 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
@@ -188,7 +188,7 @@ static void rvu_sso_ggrp_taq_flush(struct rvu *rvu, u16 pcifunc, int lf,
 	reg = rvu_read64(rvu, blkaddr, SSO_AF_HWGRPX_TAQ_THR(lf));
 	while ((reg >> 48) & 0x7FF) {
 		rvu_write64(rvu, blkaddr,
-			    SSO_AF_BAR2_ALIASX(lf, SSO_LF_GGRP_OP_ADD_WORK1),
+			    SSO_AF_BAR2_ALIASX(slot, SSO_LF_GGRP_OP_ADD_WORK1),
 			    0x1 << 3);
 get_work:
 		rvu_write64(rvu, ssow_blkaddr,
-- 
2.31.1

